### PR TITLE
T22 - Swap Indices

### PR DESCRIPTION
**Pull request by arivazhagan-tmm**

### CODE REVIEW

File: T22/Program.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿using static System.Console;
+
+internal class Program {
+
+   // Swaps digits of a number based on user given index values.
+   private static void Main (string[] args) {
```
Comment (varalakshmi-imm) [inline T22/Program.cs:6]: Can remove unused args

File: T22/Program.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+         var (m, n) = (num[j], num[i]);
+         Write ($"Result : ");
+         for (int k = 0; k < len; k++) {
+            if (k == i) Print (m);
+            else if (k == j) Print (n);
+            else Write ($" {num[k]}");
+         }
+         WriteLine ();
+         WriteLine ();
+         Write ("Do you want to continue? (Y/N) : ");
```
Comment (varalakshmi-imm) [inline T22/Program.cs:23]:
```suggestion
         Write ("Do you want to continue? (Y/N): ");

```

File: T22/Program.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+   // Swaps digits of a number based on user given index values.
+   private static void Main (string[] args) {
+      var rand = new Random ();
+      while (true) {
+         var num = rand.Next (1000, 100000).ToString ();
+         var len = num.Length;
+         WriteLine ($"The number is {num}.");
+         GetResponse ("Enter the first index\t: ", len, out int i);
+         GetResponse ("Enter the second index\t: ", len, out int j);
+         var (m, n) = (num[j], num[i]);
```
Comment (varalakshmi-imm) [inline T22/Program.cs:14]: Have a separate Swap method which takes the number and returns the swapped number.
If it is a char array you can do tuple swap.

File: T22/Program.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+         WriteLine ();
+      }
+   }
+
+   // Getting a valid numeric value from the user for the given prompt.
+   static void GetResponse (string prompt, int max, out int i) {
+      Write (prompt);
+      while (!int.TryParse (ReadLine (), out i) || i >= max) {
+         ForegroundColor = ConsoleColor.Red;
+         var message = i >= max ? $"\tIndex should be less than {max}" : "\tPlease enter valid numeric value!";
```
Comment (varalakshmi-imm) [inline T22/Program.cs:34]: -1 is not handled for index

**Review by varalakshmi-imm — CHANGES_REQUESTED** [review]

File: T22/Program.cs (hunk 3 above)
Comment (arivazhagan-tmm) [reply T22/Program.cs]: I have simplified the code without using swap logic.

Comment (varalakshmi-imm) [reply T22/Program.cs]: I think this assignment expects to have a method to swap.

**Review by varalakshmi-imm — CHANGES_REQUESTED** [review]
Implement a Swap method

**Review by varalakshmi-imm — CHANGES_REQUESTED** [review]
I guess I didn't make it clear.
Swap method should take the number and the indices and return the new number.

File: T22/Program.cs [hunk 5]
```diff
@@ -39,18 +40,8 @@ internal class Program {
...
-
-   // Swaps given two numbers and store them into their actual address.
-   static (int a, int b) Swap (int a, int b) {
-      int tmp = a;
-      a = b;
-      b = tmp;
-      return (a, b);
+   // Swaps the values present at the given indices in the given array .
+   static void Swap (char[] chars, int index1, int index2) {
+      (chars[index1], chars[index2]) = (chars[index2], chars[index1]);
```
Comment (varalakshmi-imm) [inline T22/Program.cs:45]: image

@arivazhagan-tmm Always use expression body or single line methods

**Review by varalakshmi-imm — APPROVED** [review]

**Pull request closed without merging**